pymc/sampling.py

```diff
@@ -1727,6 +1727,7 @@ def sample_posterior_predictive(
             if rv not in vars_to_sample
             and rv in model.named_vars.values()
             and not isinstance(rv, (Constant, SharedVariable))
+            and rv.name in _trace[0]  # Only accept tensors that have values in the trace points
```

Comment:
This and the else block are pretty confusing, and it looks like they should be doing the same. 

Can't we just save the list of variables for which we have known values (regardless if it's in a multitrace or idata) and then use the same logic to pick the input `RV`s and their trace values?

Reply:
Yes, we could do that too. I didn't want to change the code base too much though, and this filtering was all that was necessary to fix the problem with this particular posterior predictive sampling

Reply:
Cleaning it up would help maintaining things down the road. Do you mind doing it in this PR or do you prefer opening a separate issue?

Reply:
No problem, I’ll do it here